### PR TITLE
test: Implement Unit Tests for FormLoaderStatus Component

### LINKED ISSUE #10: Implement Unit Tests for FormLoaderStatus Component
This issue is focused on implementing unit tests for the FormLoaderStatus component within our project. Unit testing is a crucial aspect of software development that helps ensure the reliability and correctness of individual code units, such as components.

**Goals:**

- Create comprehensive unit tests for the FormLoaderStatus component to validate its functionality.
- Verify that the FormLoaderStatus component behaves as expected.
- Ensure that any changes or updates to the FormLoaderStatus component in the future do not introduce regressions.

**Tasks:**

1. Write unit tests that cover different aspects of the FormLoaderStatus component (e.g., rendering spinner when loading, show error messages).
2. Validate the accessibility of the FormLoaderStatus component, ensuring it meets accessibility standards.
3. Ensure that the tests provide sufficient code coverage for the Button component.

**Guidelines:**

- Follow best practices for unit testing.
- Write clear and descriptive test cases that cover both typical and edge cases.
- Pay attention to code quality, readability, and maintainability when writing tests.

### PR DESCRIPTION
**Pull request by vibh1103**
Create a new file index.spec.tsx in the Formloader status contains the testcase to cover various test case

Closes #10

### CODE REVIEW

File: src/presentation/components/FormLoaderStatus/index.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { RenderResult, render } from '@testing-library/react';
+import { FormLoaderStatus } from '@/presentation/components';
+import Context from '@/presentation/hooks/form';
+
+const renderComponent = (isLoading: boolean, mainError: string): RenderResult => {
```
Comment (rubemfsv) [inline src/presentation/components/FormLoaderStatus/index.spec.tsx]: I recommend to use the name "makeSut" to this const, due to the pattern in the other test scenarios

File: src/presentation/components/FormLoaderStatus/index.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const isLoading = false;
+    const mainError = '';
+    const { queryByTestId } = renderComponent(isLoading, mainError);
+    const spinner = queryByTestId('spinner');
+    expect(spinner).toBeNull();
+  });
+
+  test('Shows error message when mainError is provided', () => {
+    const isLoading = false;
+    const mainError = 'An error occurred';
```
Comment (rubemfsv) [inline src/presentation/components/FormLoaderStatus/index.spec.tsx]: You can use faker

File: src/presentation/components/FormLoaderStatus/index.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const mainError = '';
+    const { getByTestId } = renderComponent(isLoading, mainError);
+    const spinner = getByTestId('spinner');
+    expect(spinner).toBeInTheDocument();
+  });
+
+  test('Does not render spinner when isLoading is false', () => {
+    const isLoading = false;
+    const mainError = '';
+    const { queryByTestId } = renderComponent(isLoading, mainError);
```
Comment (rubemfsv) [inline src/presentation/components/FormLoaderStatus/index.spec.tsx]: you can attribute an empty string as default to mainError in line 6, so you dont need to declare in every test, and I recommend you to use getByTestId as the other scenarios

**Review by rubemfsv — CHANGES_REQUESTED** [review]

File: src/presentation/components/FormLoaderStatus/index.spec.tsx (hunk 1 above)
Comment (vibh1103) [reply src/presentation/components/FormLoaderStatus/index.spec.tsx]: Done

File: src/presentation/components/FormLoaderStatus/index.spec.tsx (hunk 2 above)
Comment (vibh1103) [reply src/presentation/components/FormLoaderStatus/index.spec.tsx]: Done

File: src/presentation/components/FormLoaderStatus/index.spec.tsx (hunk 3 above)
Comment (vibh1103) [reply src/presentation/components/FormLoaderStatus/index.spec.tsx]: empty string part is done for `mainError` and `isLoading`.  

I can use `getByTestId` but it throws error when the element is not rendered but on other side `queryByTestId` doesn't throw error. Please confirm me once if i have to change ?

**Review by rubemfsv — APPROVED** [review]

**Comment by rubemfsv** [comment]
Thanks for tour support, @vibh1103!
